Sources/SwiftInspectorVisitors/TypeSyntaxVisitor.swift

```diff
@@ -247,7 +247,7 @@ public struct TypeProperties: Hashable {
   /// The name of the type.
   public let name: String
   /// The properties on this type
-  public let properties: [PropertyData]
+  public let properties: Set<PropertyData>
```

Comment:
Makes sense that each property should be unique. Curious if/how you envision looking up the property data? I would think a dictionary might be more interesting so we could do a fast property name -> property data lookup.

Reply:
Currently (and before) we need to iterate the sequence to find an individual property.

A set seemed like a nice improvement since the code was [already](https://github.com/fdiaz/SwiftInspector/blob/2a901892629c14e828e165c58bc530d505494a30/Sources/SwiftInspectorAnalyzers/PropertyAnalyzer.swift) casting the properties to a set.

A dictionary is an interesting idea but a complication is that you can have a static and instance property of the same name, so the key can't be only the property name.

It's hard for me to know what a the most common workflow for this data would be: looking at all properties or finding a specific property.

Even though finding a specific property is O(N) I think it's reasonable to assume that N will be fairly small in this case.